### PR TITLE
chore: cleanup empty default values

### PR DESCRIPTION
**Pull request by iliapolo**
Use `undefined` instead of empty collections where applicable. This makes the manifest less verbose and easier on the eyes.

### CODE REVIEW

File: test/pod.test.ts [hunk 1]
```diff
@@ -4,6 +4,17 @@ import * as kplus from '../src';
 import { DockerConfigSecret, FsGroupChangePolicy, Probe } from '../src';
 import * as k8s from '../src/imports/k8s';
 
+test('deafults', () => {
```
Comment (Chriscbr) [inline test/pod.test.ts]:
```suggestion
test('defaults', () => {
```

File: src/utils.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 export function undefinedIfEmpty<T>(obj: T): T | undefined {
   if (typeof(obj) === 'string' && obj === '') { return undefined; }
   if (Array.isArray(obj) && obj.length === 0) { return undefined; }
-  if (typeof(obj) === 'object' && Object.keys(obj).length === 0) { return undefined; }
+  if (typeof(obj) === 'object' && Object.keys(obj).length === 0 || Object.values(obj).filter(x => x).length === 0) { return undefined; }
```
Comment (Chriscbr) [inline src/utils.ts]: Can we add parentheses to make the intended order of ops here clearer? A unit test would also help here

**Review by Chriscbr — APPROVED** [review]

File: src/utils.ts (hunk 2 above)
Comment (iliapolo) [reply src/utils.ts]: Added parentheses. As for unit test, this is already covered by our snapshot tests.

**Comment by iliapolo** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|k8s-21/main|[](https://github.com/cdk8s-team/cdk8s-plus/pull/708)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Comment by iliapolo** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|k8s-20/main|[](https://github.com/cdk8s-team/cdk8s-plus/pull/709)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**